pinc/RoundDescriptor.inc

```diff
@@ -199,6 +199,25 @@ class Round extends Stage
         // Give them the lowest honorific.
         return $honorific;
     }
+
+    // -----------
+
+    public function validate_user_can_access($user)
+    {
+        global $code_url;
+
+        $uao = $this->user_access($user);
+        if (!$uao->can_access) {
+            $message = sprintf(
+                // TRANSLATORS: %1$s is the round name, %2$s is the round ID (eg: P1)
+                _('You have not yet been cleared to work on projects in %1$s (%2$s). Please visit <a href="%3$s">the %2$s home</a> to find out what happens in this round and how you can qualify to work in it.'),
```

Comment:
This message contains html so would not be ideal for showing in an alert box. It would presumably work in a <dialog> element but we cannot use this yet.
In the pesent code, if it is shown in the abort page it is wrapped in html_safe() which is also not ideal. (It cannot happen normally, only by e.g. doing a page in F2, then revoking your access to F2 and then following the link to the page you did in the project page.)
Since it is quite acceptable to look at the project page of a project in a round you are not qualified to work in, perhaps we should consider making a message without html and catching this exception in the project page and adding the second sentence there.

Reply:
Done this.